### ISSUE TITLE
grafana-cli plugin install: infinite install loop with pluginUrl and dependencies

### ISSUE THREAD

**Issue by robx** [grafana/grafana#26303]
If `grafana-cli plugin install` is passed a URL to a plugin with dependencies, the tool will loop because it tries to download the dependencies from the same URL.

**What happened**:

```
$ grafana-cli --pluginUrl=http://localhost:8000/test-plugin.zip install my-test-plugin
✔ Installed  successfully 
installing  @ 
from: http://localhost:8000/test-plugin.zip
into: /g/grafana/data/plugins

✔ Installed  successfully 
installing  @ 
from: http://localhost:8000/test-plugin.zip
into: /g/grafana/data/plugins

...
```

**What you expected to happen**:

Stopping with an error when encountering dependencies in a plugin installed from URL.

(Maybe, ideally, installing dependencies from grafana.com.)

**How to reproduce it (as minimally and precisely as possible)**:

1. create a plugin with a non-empty dependencies list
2. host the plugin
3. call `grafana-cli --pluginUrl=\<plugin-url\> install plugin some-plugin-name`

**Environment**:
- Grafana version: current master

**Comment by ivanahuckova** [comment]
@ryantxu Can you please check this issue out?

**Comment by cobolbaby** [comment]
Same Question: 

```
✔ Installed anodot-datasource successfully 
installing anodot-panel @ 1.0.0
from: https://grafana.com/api/plugins/anodot-panel/versions/1.0.0/download
into: /grafana-plugins

✔ Installed anodot-panel successfully 
installing anodot-datasource @ 1.0.0
from: https://grafana.com/api/plugins/anodot-datasource/versions/1.0.0/download
into: /grafana-plugins
```

It seems that the md5 checksum is incorrect.

**Comment by gooduling** [comment]
@marcusolsson Hi, looks like in our new-born plugin we have the same issue now. Are there any ready solutions? https://github.com/anodot/grafana-panel/issues/1

**Comment by wbrowne** [comment]
Fixed with #33274

**Comment by cobolbaby** [comment]
@wbrowne  Is this fix only for the latest 8.x branch?

**Comment by wbrowne** [comment]
@cobolbaby Yes, this change is available from v8.0.0
